vm/systemSmartContracts/staking.go

```diff
@@ -565,12 +565,15 @@ func (s *stakingSC) unStakeAtEndOfEpoch(args *vmcommon.ContractCallInput) vmcomm
 		return vmcommon.Ok
 	}
 
-	if !registrationData.Staked {
-		log.Debug("stakingSC.unStakeAtEndOfEpoch: cannot unStake node which was already unStaked")
+	if !registrationData.Staked && !registrationData.Waiting {
```

Comment:
the method unStakeAtEndOfEpoch has some similarities to the unStake: multiple checks are the same in both, the values set in the registrationData are the same. Could be refactored maybe in a future PR

Reply:
yes. it has. first we should remove the flags. after that see what is common on the new version